docs/grammar.md

```diff
@@ -0,0 +1,90 @@
+Syntax
+------
```

Comment:
I wonder if we should include a small example for each documented construct? That usually helps readers grasp the construct quicker than just reading the abstract description.

Reply:
I was having difficulties coming up with useful examples for lookahead and cut operators.